autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -0,0 +1,533 @@
+import logging.handlers
+from multiprocessing.queues import Queue
+import time
+from typing import Any, Dict, List, Optional, Tuple, Union
+import warnings
+
+from ConfigSpace import Configuration
+
+import numpy as np
+
+from smac.tae import StatusType
+
+from sklearn.base import BaseEstimator
+from sklearn.dummy import DummyClassifier, DummyRegressor
+from sklearn.ensemble import VotingClassifier, VotingRegressor
+
+from autoPyTorch.constants import (
+    CLASSIFICATION_TASKS,
+    IMAGE_TASKS,
+    MULTICLASS,
+    REGRESSION_TASKS,
+    STRING_TO_OUTPUT_TYPES,
+    STRING_TO_TASK_TYPES,
+    TABULAR_TASKS,
+)
+from autoPyTorch.datasets.base_dataset import BaseDataset
+from autoPyTorch.evaluation.utils import (
+    convert_multioutput_multiclass_to_multilabel
+)
+from autoPyTorch.pipeline.components.training.metrics.base import autoPyTorchMetric
+from autoPyTorch.pipeline.components.training.metrics.utils import (
+    calculate_score,
+)
+from autoPyTorch.pipeline.image_classification import ImageClassificationPipeline
+from autoPyTorch.pipeline.tabular_classification import TabularClassificationPipeline
+from autoPyTorch.pipeline.tabular_regression import TabularRegressionPipeline
+from autoPyTorch.utils.backend import Backend
+from autoPyTorch.utils.logging_ import get_named_client_logger, PicklableClientLogger
+from autoPyTorch.utils.pipeline import get_dataset_requirements
+
+__all__ = [
+    'AbstractEvaluator',
+    'fit_and_suppress_warnings'
+]
+
+
+# TODO: make dummypipeline or model for autopytorch
+class MyDummyClassifier(DummyClassifier):
+    def __init__(self, config: Configuration,
```

Comment:
Could we write some explaining docstrings in this module?

Reply:
Yes I'll do that